lib/Agrammon/Preprocessor.pm6

```diff
@@ -64,18 +80,18 @@ sub preprocess(Str $source, %options --> Str) is export {
         if $content.starts-with('?') {
             @result-lines.push('');
             given $content {
-                when /^ '?if' \h+ [$<negate>='!']? <option=.ident> \h* $/ {
-                    my $enabled = $<negate> ?? .not !! .so given %options{~$<option>};
+                when /^ '?if' $<parts>=(\h+ [$<negate>='!']? <option=.ident>)+ % [\h+ $<op>=< and or >] \h* $/ {
+                    my $enabled = process-match($<parts>, $<negate>, %options, $<op>, $<option>, $number);
                     my $matched = $enabled;
                     my $start-line = $number + 1;
                     @open.push(OpenDirective.new(:$start-line, :$enabled, :$matched, :accept-elsif));
                 }
-                when /^ '?elsif' \h+ [$<negate>='!']? <option=.ident> \h* $/ {
+                when /^ '?elsif' $<parts>=(\h+ [$<negate>='!']? <option=.ident>)+ % [\h+ $<op>=< and or >] \h* $/ {
                     if @open.pop -> $prev-part {
                         unless $prev-part.accept-elsif {
                             die X::Agrammon::Preprocessor::ElsifAfterElse.new: :line($number + 1);
                         }
-                        my $enabled = so !$prev-part.matched && ($<negate> ?? .not !! .so given %options{~$<option>});
+                        my $enabled = so !$prev-part.matched && process-match($<parts>, $<negate>, %options, $<op>, $<option>, $number);
```

Comment:
It's a nitpick but the `so` is not needed here given `!` and `process-match` both return booleans.

Reply:
Removed.